src/Microsoft.ML.AutoML/AutoMLExperiment/IDatasetManager.cs

```diff
@@ -12,34 +15,83 @@ public interface IDatasetManager
 
-        IDataView ValidateDataset { get; set; }
+        IDataView LoadValidateDataset(MLContext context, TrialSettings? settings);
     }
 
     internal class TrainValidateDatasetManager : IDatasetManager, ITrainValidateDatasetManager
     {
-        public IDataView TrainDataset { get; set; }
+        private ulong _rowCount;
+        private IDataView _trainDataset;
+        private readonly IDataView _validateDataset;
+        private readonly string _subSamplingKey = "TrainValidateDatasetSubsamplingKey";
+        private bool _isInitialized = false;
+        public TrainValidateDatasetManager(IDataView trainDataset, IDataView validateDataset, string? subSamplingKey = null)
+        {
+            _trainDataset = trainDataset;
+            _validateDataset = validateDataset;
+            _subSamplingKey = subSamplingKey ?? _subSamplingKey;
+        }
+
+        public string SubSamplingKey => _subSamplingKey;
+
+        /// <summary>
+        /// Load Train Dataset. If <see cref="TrialSettings.Parameter"/> contains <see cref="_subSamplingKey"/> then the train dataset will be subsampled.
+        /// </summary>
+        /// <param name="context">MLContext.</param>
+        /// <param name="settings">trial settings. If null, return entire train dataset.</param>
+        /// <returns>train dataset.</returns>
+        public IDataView LoadTrainDataset(MLContext context, TrialSettings? settings)
+        {
+            if (!_isInitialized)
+            {
+                InitializeTrainDataset(context);
+                _isInitialized = true;
+            }
+            var trainTestSplitParameter = settings?.Parameter.ContainsKey(nameof(TrainValidateDatasetManager)) is true ? settings.Parameter[nameof(TrainValidateDatasetManager)] : null;
+            if (trainTestSplitParameter is Parameter parameter)
+            {
+                var subSampleRatio = parameter.ContainsKey(_subSamplingKey) ? parameter[_subSamplingKey].AsType<double>() : 1;
+                if (subSampleRatio < 1.0)
+                {
+                    var subSampledTrainDataset = context.Data.TakeRows(_trainDataset, (long)(subSampleRatio * _rowCount));
+                    return subSampledTrainDataset;
+                }
+            }
 
-        public IDataView ValidateDataset { get; set; }
+            return _trainDataset;
+        }
+
+        public IDataView LoadValidateDataset(MLContext context, TrialSettings? settings)
+        {
+            return _validateDataset;
+        }
+
+        private void InitializeTrainDataset(MLContext context)
+        {
+            _rowCount = DatasetDimensionsUtil.CountRows(_trainDataset, ulong.MaxValue);
+            _trainDataset = context.Data.ShuffleRows(_trainDataset);
+        }
     }
 
     internal class CrossValidateDatasetManager : IDatasetManager, ICrossValidateDatasetManager
     {
-        public IDataView Dataset { get; set; }
+        public IDataView? Dataset { get; set; }
```

Comment:
Do you need this to be a nullable type? Can an IDataView directly have null as a value that you can just check?

Reply:
This should not be nullable. I'll make the change